### PR TITLE
Support psalm v5

### PR DESCRIPTION
**Pull request by veewee**
Psalm v5 is out!

This PR contains:

* A test-run of https://github.com/php-standard-library/psalm-plugin/pull/9
* Fixes for latest psalm

Awaiting:

* [x] Validation of linked psalm-plugin issue
* [x] roave/infection-static-analysis-plugin
  * https://github.com/Roave/infection-static-analysis-plugin/pull/461

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -23,10 +23,9 @@
         "friendsofphp/php-cs-fixer": "^3.4.0",
         "phpbench/phpbench": "^1.2.3",
         "phpunit/phpunit": "^9.5.16",
-        "vimeo/psalm": "^4.21.0",
-        "php-standard-library/psalm-plugin": "^2.0.2",
-        "php-coveralls/php-coveralls": "^2.5.2",
-        "roave/infection-static-analysis-plugin": "^1.23.0"
+        "vimeo/psalm": "^5.0.0",
+        "php-standard-library/psalm-plugin": "dev-psalmv5",
```
Comment (veewee) [inline composer.json]: Temporary changes to test the psalm-plugin changes. Will revert once the changes in psalm-plugin are released

File: src/Psl/Filesystem/exists.php [hunk 2]
```diff
@@ -9,9 +9,7 @@
 /**
  * Check whether $node exists.
  *
- * @param string $node Path, absolute or relative to the current working directory.
- *
- * @psalm-assert-if-true non-empty-string $node
```
Comment (veewee) [inline src/Psl/Filesystem/exists.php:14]: The assert here causes NoValue errors all over the place. I think it's better to make it non-empty to start with

File: src/Psl/Collection/MutableVector.php [hunk 3]
```diff
@@ -623,6 +623,8 @@ public function slice(int $start, ?int $length = null): MutableVector
      *                                         `MutableVector` split into chunks of the given size.
      *
      * @psalm-mutation-free
+     *
+     * @psalm-suppress LessSpecificImplementedReturnType - I don't see how this one is less specific than its inherited.
```
Comment (veewee) [inline src/Psl/Collection/MutableVector.php:627]: Maybe you know?

**Review by azjezz — DISMISSED** [review]
LGTM 🚢

File: src/Psl/Filesystem/exists.php (hunk 2 above)
Comment (azjezz) [reply src/Psl/Filesystem/exists.php:14]: can you open a bug for psalm about this? if able to replicate it.

File: src/Psl/Collection/MutableVector.php (hunk 3 above)
Comment (azjezz) [reply src/Psl/Collection/MutableVector.php:627]: 🤷‍♂️ there's actually more of these even with psalm v4

File: src/Psl/Filesystem/exists.php (hunk 2 above)
Comment (veewee) [reply src/Psl/Filesystem/exists.php:14]: It has been reported already: https://github.com/vimeo/psalm/issues/8659

**Comment by veewee** [comment]
Changed base branch to 2.3.x (which needs to become the default one), so need a re-accept

**Review by azjezz — APPROVED** [review]

**Comment by azjezz** [comment]
updated default branch 👍

**Pull request merged**